### PR TITLE
fix(DatePicker): disabled style is overridden in Form hover

### LINKED ISSUE #44780: DatePicker disabled 样式在 Form 下消失
### Reproduction link

[https://stackblitz.com/edit/react-eevocz?file=demo.tsx](https://stackblitz.com/edit/react-eevocz?file=demo.tsx)

### Steps to reproduce

1. 点击提交
2. 鼠标 hover 到 DatePicker 内

### What is expected?

disabled 保持不变

### What is actually happening?

disabled 样式失效

| Environment | Info |
| --- | --- |
| antd | 4.24.14 |
| React | latest |
| System | mac |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix: #44780 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | disabled style is overridden in Form hover |
| 🇨🇳 Chinese |   `disabled` 样式在 Form hover 场景下被覆盖      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 1595d09</samp>

Fixed a date picker hover style bug by using a more specific selector in `components/date-picker/style/status.less`. This was part of a larger effort to enhance the date picker's accessibility and usability.

### <samp>🤖 Generated by Copilot at 1595d09</samp>

* Fix hover style of disabled date picker ([link](https://github.com/ant-design/ant-design/pull/44779/files?diff=unified&w=0#diff-ea8f77385ec1a97927b9a52977e4fb1cad59d2fc26891fd510eef59114d8f936L14-R14))

### CODE REVIEW

File: components/date-picker/style/status.less [hunk 1]
```diff
@@ -11,7 +11,7 @@
 ) {
   &.@{picker-prefix-cls} {
     &,
-    &:not([disabled]):hover {
+    &:not(.@{picker-prefix-cls}):hover {
```
Comment (Wxh16144) [inline components/date-picker/style/status.less:14]: 是不是写错了？

```suggestion
    &:not(.@{picker-prefix-cls}-disabled):hover {
```

Comment (MadCcc) [inline components/date-picker/style/status.less:14]: 这看起来不太对？编译后就变成了 
```css
.cls.picker:not(.picker):hover {}
```

File: components/_util/ActionButton.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ const ActionButton: React.FC<ActionButtonProps> = (props) => {
   };
 
   React.useEffect(() => {
-    let timeoutId: NodeJS.Timer | null = null;
+    let timeoutId: ReturnType<typeof setTimeout> | null = null;
```
Comment (kiner-tang) [inline components/_util/ActionButton.tsx:33]: 建议把这个类型抽离到全局吧，感觉没必要所有地方都用 ReturnType 动态计算

File: components/date-picker/style/status.less (hunk 1 above)
Comment (crazyair) [reply components/date-picker/style/status.less]: 你们看的太快了。。。

修改了

File: components/_util/ActionButton.tsx (hunk 2 above)
Comment (Wxh16144) [reply components/_util/ActionButton.tsx:33]: 其实我都感觉原来的  `NodeJS.Timer` 可以不用改

Comment (crazyair) [reply components/_util/ActionButton.tsx:33]: 有放全局 types 的文件吗

Comment (MadCcc) [reply components/_util/ActionButton.tsx:33]: 不稳定，ReturnType 是一定不会挂的

File: components/_util/type.ts [hunk 3]
```diff
@@ -11,3 +11,5 @@ export type ElementOf<T> = T extends (infer E)[] ? E : T extends readonly (infer
 
 /** https://github.com/Microsoft/TypeScript/issues/29729 */
 export type LiteralUnion<T extends U, U> = T | (U & {});
+
+export type setTimeoutResult = ReturnType<typeof setTimeout> | null;
```
Comment (kiner-tang) [inline components/_util/type.ts:15]:
```suggestion
export type TimeoutId = ReturnType<typeof setTimeout> | null;
```

File: components/_util/ActionButton.tsx [hunk 4]
```diff
@@ -3,6 +3,7 @@ import * as React from 'react';
 import Button from '../button';
 import type { ButtonProps, LegacyButtonType } from '../button/button';
 import { convertLegacyProps } from '../button/button';
+import type { setTimeoutResult } from './type';
```
Comment (afc163) [inline components/_util/ActionButton.tsx:6]: 这个 PR 里为啥改这个？

Comment (crazyair) [reply components/_util/ActionButton.tsx]: 执行 `tsc` 会报错

File: package.json [hunk 5]
```diff
@@ -335,7 +335,7 @@
   "bundlesize": [
     {
       "path": "./dist/antd.min.js",
-      "maxSize": "285 kB"
+      "maxSize": "286 kB"
```
Comment (yoyo837) [inline package.json:338]: 尽量卡死一点, 285.5 如何? 😂

Comment (crazyair) [reply package.json:338]: https://github.com/ant-design/ant-design/pull/39267 参考的这个增加的幅度

File: components/_util/ActionButton.tsx (hunk 4 above)
Comment (afc163) [reply components/_util/ActionButton.tsx]: 我看 4.x-stable 分支上没报错？

Comment (crazyair) [reply components/_util/ActionButton.tsx]: 这是第一个 commit，还没改这里的报错 https://github.com/ant-design/ant-design/actions/runs/6155604001/job/16702821421

Comment (afc163) [reply components/_util/ActionButton.tsx]: 你可能不是从最新的 4.x-stable 分支上切出来的？

Comment (afc163) [reply components/_util/ActionButton.tsx]: 你现在可以 revert 掉这部分改动试试。

Comment (crazyair) [reply components/_util/ActionButton.tsx]: 可以了

**Review by afc163 — APPROVED** [review]

**Pull request merged**